src/test/java/org/prebid/server/bidder/adtrgtme/AdtrgtmeBidderTest.java

```diff
@@ -0,0 +1,339 @@
+                                .ext(mapper.valueToTree(ExtPrebid.of(null, ExtImpAdrino.of("test"))))))).build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = adtrgtmeBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(2);
+    }
+
+    @Test
+    public void makeBidsShouldReturnEmptyListIfBidResponseIsNull() throws JsonProcessingException {
+        // given
+        final BidRequest bidRequest = givenBidRequest(impBuilder -> impBuilder.id("123"));
+        final BidderCall<BidRequest> httpCall = givenHttpCall(
+                bidRequest,
+                mapper.writeValueAsString(null));
+
+        // when
+        final Result<List<BidderBid>> result = adtrgtmeBidder.makeBids(httpCall, bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeBidsShouldReturnBidsFromDifferentSeatBidsInResponse() throws JsonProcessingException {
+        // given
+        final String response = mapper.writeValueAsString(BidResponse.builder()
+                .seatbid(asList(
+                        SeatBid.builder()
+                                .bid(singletonList(Bid.builder().id("bidId1").impid("impId1").build()))
+                                .build(),
+                        SeatBid.builder()
+                                .bid(singletonList(Bid.builder().id("bidId2").impid("impId2").build()))
+                                .build()))
+                .build());
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(asList(Imp.builder().id("impId1").banner(Banner.builder().build()).build(),
+                        Imp.builder().id("impId2").banner(Banner.builder().build()).build()))
+                .build();
+
+        final BidderCall<BidRequest> httpCall = givenHttpCall(bidRequest, response);
+
+        // when
+        final Result<List<BidderBid>> result = adtrgtmeBidder.makeBids(httpCall, bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue())
+                .extracting(BidderBid::getBid)
+                .extracting(Bid::getId).containsExactly("bidId1", "bidId2");
+    }
+
+    @Test
+    public void makeBidsShouldReturnEmptyBidderWithErrorWhenResponseCanNotBeParsed() {
+        // given
+        final BidderCall<BidRequest> httpCall = givenHttpCall(null, "{");
+
+        // when
+        final Result<List<BidderBid>> result = adtrgtmeBidder.makeBids(httpCall, BidRequest.builder().build());
+
+        // then
+        assertThat(result.getErrors()).hasSize(1).element(0).satisfies(bidderError -> {
+            assertThat(bidderError.getType()).isEqualTo(BidderError.Type.bad_server_response);
+            assertThat(bidderError.getMessage()).startsWith("Failed to decode:");
+        });
```

Comment:
better check with smth like `allSatisfy` instead of `.hasSize(1).element(0)` or with `containsExactly`